Mathlib/Topology/Constructions.lean

```diff
@@ -1212,6 +1212,32 @@ theorem frontier_inter_open_inter {s t : Set X} (ht : IsOpen t) :
     ht.isOpenMap_subtype_val.preimage_frontier_eq_frontier_preimage continuous_subtype_val,
     Subtype.preimage_coe_self_inter]
 
+open scoped Set.Notation in
+lemma IsOpen.preimage_val {s t : Set X} (ht : IsOpen t) : IsOpen (s ↓∩ t) := isOpen_induced ht
+
+open scoped Set.Notation in
+lemma IsClosed.preimage_val {s t : Set X} (ht : IsClosed t) : IsClosed (s ↓∩ t) := by
+  rw [← isOpen_compl_iff] at ht ⊢
+  exact ht.preimage_val
```

Comment:
```suggestion
lemma IsClosed.preimage_val {s t : Set X} (ht : IsClosed t) : IsClosed (s ↓∩ t) := 
  ht.preimage continuous_subtype_val
```

I'm tempted to suggest the same for the lemma above: morally this is just continuity of the inclusion.